src/Microsoft.Data.Sqlite/SqliteConnection.cs

```diff
@@ -94,12 +111,12 @@ public override void Open()
                 throw new InvalidOperationException(Strings.OpenRequiresSetConnectionString);
             }
 
-            var flags = Constants.SQLITE_OPEN_READWRITE | Constants.SQLITE_OPEN_CREATE;
+            var flags = Constants.SQLITE_OPEN_READWRITE | Constants.SQLITE_OPEN_CREATE | Constants.SQLITE_OPEN_URI;
             flags |= (ConnectionStringBuilder.Cache == SqliteConnectionCacheMode.Shared) ? Constants.SQLITE_OPEN_SHAREDCACHE : Constants.SQLITE_OPEN_PRIVATECACHE;
 
             var path = ConnectionStringBuilder.DataSource;
 
-            if (!path.Equals(":memory:", StringComparison.OrdinalIgnoreCase))
+            if (!IsInMemory(path))
```

Comment:
So, instead receive ConnectionStringBuilder params and translate to their equivalent `SQLITE_OPEN_*` [flags](https://www.sqlite.org/c3ref/c_open_autoproxy.html)?